File: chat-backend/src/main/java/com/chat/GeneratePassword.java
package com.chat;

import java.security.MessageDigest;
import java.util.Base64;

/**
 * 生成正确的加密密码
 */
public class GeneratePassword {

    private static final String SALT = "chat-app-salt-key";

    public static String encodePassword(String password) {
        try {
            String saltedPassword = password + SALT;
            MessageDigest md = MessageDigest.getInstance("SHA-256");
            byte[] hashedBytes = md.digest(saltedPassword.getBytes());
            return Base64.getEncoder().encodeToString(hashedBytes);
        } catch (Exception e) {
            throw new RuntimeException("密码加密失败", e);
        }
    }

    public static void main(String[] args) {
        String password = "test123";
        String encoded = encodePassword(password);

        System.out.println("=================================");
        System.out.println("原始密码: " + password);
        System.out.println("加密后:   " + encoded);
        System.out.println("=================================");
        System.out.println("");
        System.out.println("请在 MySQL 中执行以下命令:");
        System.out.println("");
        System.out.println("UPDATE t_user SET password = '" + encoded + "' WHERE username IN ('admin', 'user1', 'user2');");
        System.out.println("");
        System.out.println("然后重试登录");
    }
}

File: generate-password.js
#!/usr/bin/env node

const crypto = require('crypto');
const base64 = require('buffer').Buffer;

function encodePassword(password) {
  const salt = 'chat-app-salt-key';
  const saltedPassword = password + salt;
  const hashedBytes = crypto.createHash('sha256').update(saltedPassword).digest();
  return hashedBytes.toString('base64');
}

// 生成 test123 的加密密码
const password = 'test123';
const encoded = encodePassword(password);

console.log('原始密码:', password);
console.log('加密后:', encoded);
console.log('');
console.log('SQL 更新语句:');
console.log(`UPDATE t_user SET password = '${encoded}' WHERE username IN ('admin', 'user1', 'user2');`);
